File: assets/static/blog/dsa/data_structures/queues/implementation.go
package main
import "fmt"

type Queue struct {
  data []interface{}
  size int
}

func (q *Queue) Enqueue(val interface{}) {
  q.data = append(q.data, val)
  q.size += 1
}

func (q *Queue) Dequeue() interface{} {
  if q.size == 0 {return nil}
  val := q.data[0]
  q.data = q.data[1:]
  q.size -= 1
  return val
}


func main() {
  q := Queue{}
  q.Enqueue(1)
  q.Enqueue(2)
  q.Enqueue(3)

  fmt.Println(q.data, "|", q.size)
  val := q.Dequeue()
  val = q.Dequeue()
  val = q.Dequeue()
  fmt.Println(val)
  fmt.Println(q.data, "|", q.size)
}

File: assets/static/blog/dsa/data_structures/queues/implementation.py
class Queue:
    def __init__(self, data=None):
        self.data = data or []

    def __repr__(self):
        return str(self.data)

    def enqueue(self, item):
        """Insert an item at the start of the array"""
        self.data.insert(0, item)

    def dequeue(self):
        """Remove an element from the end of the array and return it"""
        return self.data.pop()

    def is_empty(self):
        return self.data == []

    def size(self):
        return len(self.data)
